src/layout/RepeatingGroup/RepeatingGroupFocusContext.tsx

```diff
@@ -36,40 +36,48 @@ export function RepeatingGroupsFocusProvider({ children }: PropsWithChildren) {
   const elementRefs = useMemo(() => new Map<string, HTMLElement | null>(), []);
   const waitingForFocus = useRef<number | null>(null);
 
-  const { node, openForEditing } = useRepeatingGroup();
+  const { node, openForEditing, changePageToRow } = useRepeatingGroup();
   useRegisterNodeNavigationHandler((targetNode) => {
-    if (node.item.edit?.mode === 'onlyTable') {
-      // It's not possible for us to open rows for editing, so no point in doing anything here.
-      return false;
-    }
-    if (node.item.edit?.mode === 'showAll') {
-      // We're already showing all rows, so no point in doing anything here.
-      return true;
-    }
-
+    // We are a parent of the target component, and the targetChild is the target component (or a nested group
+    // containing the target component).
     let targetChild: ParentNode = targetNode;
     for (const parent of targetNode.parents()) {
       if (parent.item.id !== node.item.id) {
         targetChild = parent;
         continue;
       }
 
-      // We are a parent of the target component, and the targetChild is the target component (or a nested group
-      // containing the target component). We should most likely open the row containing targetChild for editing.
+      const row = node.item.rows.find((r) => r.items.some((i) => i.item.id === targetChild.item.id));
+
+      // If pagination is used, navigate to the correct page
+      if (node.item.pagination) {
+        if (row) {
+          changePageToRow(row.uuid);
+        } else {
+          return false;
+        }
+      }
+
+      if (node.item.edit?.mode === 'showAll' || node.item.edit?.mode === 'onlyTable') {
+        // We're already showing all nodes, so nothing further to do
+        return true;
+      }
+
+      // Check if we need to open the row containing targetChild for editing.
+      const targetChildBaseComponentId = targetChild.item.baseComponentId ?? targetChild.item.id;
       const tableColSetup =
-        (node.item.tableColumns && targetChild.item.id && node.item.tableColumns[targetChild.item.id]) || {};
+        (node.item.tableColumns && targetChildBaseComponentId && node.item.tableColumns[targetChildBaseComponentId]) ||
+        {};
 
       if (tableColSetup.editInTable || tableColSetup.showInExpandedEdit === false) {
         // No need to open rows or set editIndex for components that are rendered
         // in table (outside the edit container)
-        return false;
+        return true;
```

Comment:
I think the reason this returned false was that we can't guarantee the component is visible now, but if the component is rendered by GenericComponent somewhere on the page, that will return true. Returning true here would mean "stop looking for it, it's fine now", but there may be other navigation handlers that need to run for this to become visible (either now or at a later time with some new functionality to come). So stopping it early may be a disservice.